File: app/src/main/java/com/absinthe/kage/connect/proxy/BaseProxy.java
package com.absinthe.kage.connect.proxy;

import androidx.annotation.NonNull;

import com.absinthe.kage.connect.IProxy;
import com.absinthe.kage.device.Device;

public class BaseProxy implements IProxy {

    Device mDevice;

    @Override
    public void onDeviceConnected(@NonNull Device device) {
        mDevice = device;
    }

    @Override
    public void onDeviceDisconnected(@NonNull Device device) {

    }
}

File: app/src/main/kotlin/com/absinthe/kage/connect/proxy/BaseProxy.kt
package com.absinthe.kage.connect.proxy

import com.absinthe.kage.connect.IProxy
import com.absinthe.kage.device.Device

const val MODE_IMAGE = 0
const val MODE_AUDIO = 1
const val MODE_VIDEO = 2

open class BaseProxy : IProxy {
    var mDevice: Device? = null

    override fun onDeviceConnected(device: Device) {
        mDevice = device
    }

    override fun onDeviceDisconnected(device: Device) {
        mDevice = null
    }

    companion object {
        var CURRENT_MODE = MODE_IMAGE
    }
}